### PR TITLE
[GEOS-10221] Fix broken links in Release versions, add section on updating these links to release guide

### PR DESCRIPTION
**Pull request by tbarsballe**
[GEOS-10221](https://osgeo-org.atlassian.net/browse/GEOS-10221)

Follow-up to https://github.com/geoserver/geoserver/pull/5264

The documentation index page from that PR is now deploying automatically, and the updates can be seen on [docs.geoserver.org](https://docs.geoserver.org)

However, when adding the links for 2.14.x through 2.19.x to that PR, I neglected to test them - turns out the doc link structure changed between 2.13 and 2.14, corresponding with the doc server move. This PR fixes that error, and also adds a step to the release guide for adding a new link (with the correct structure) when a new RC branch is created.

See also https://osgeo-org.atlassian.net/browse/GEOS-10221

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: doc/en/index.html [hunk 1]
```diff
@@ -105,33 +105,33 @@ <h3>Release Versions</h3>
           </tr>
```
Comment (mprins) [inline doc/en/index.html:105]: this might be for another PR, but I was wondering if the Chinese docs should be mentioned on this page? (as of `latest`) eg. https://docs.geoserver.org/latest/zhCN/user/

**Review by mprins — APPROVED** [review]

File: doc/en/index.html (hunk 1 above)
Comment (tbarsballe) [reply doc/en/index.html:105]: Probably best as its own issue/seperate PR, yeah. I think the "best" approach would be a drop-down in the doc header allowing you to switch languages directly from a version/page, but I doubt that's feasible.

Also the Chinese docs are only really present on main right now, and only (parts) of the user guide are translated, so we'd need to determine a way of linking only the docs that do exist without looking too inconsistent. Maybe after 2.20 is released and there's actually a released version with translated docs we can add a link there.

Back when we had (partial) Spanish and French docs, I don't think they were linked anywhere either, so we don't have any defined way of representing this.
